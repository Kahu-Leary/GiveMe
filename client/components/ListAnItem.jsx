import React, { useState, useContext } from 'react'
import ImageUpload from './ImageUpload'
import { addItem } from '../apiClient'
import Nav from './Nav'
import Footer from './Footer'
import { UserContext } from './UserContext'

function ListAnItem (props) {
  const [id, setId] = useState({
    id: null
  })
  const [name, setName] = useState('')
  const [category, setCategory] = useState('')
  const [photo, setPhoto] = useState('')
  const [description, setDescription] = useState('')
  const [location, setLocation] = useState('')
  const [user, setUser] = useContext(UserContext)

  function handleSubmit (e) {
    e.preventDefault()
    addItem({ userId: user.id, name, category, photo, description, location })

    return props.history.push('/Home')
  }

  console.log('signed in id: ', user.id)

  return (
    <div className="globalBackground">
      <div className='container-footer'>
        <Nav/>
        <div className="itemForm">

          <div className='imgContainer'>
            <img className="cityImg" src='./img/city.png'></img>
          </div>

          <div className="formRight">
            <h1>List a new item</h1>
            <form onSubmit={handleSubmit}>

              <input type="hidden" name="id" value={id} onChange={event => setUser(event.target.value)}/>

              <input className="formInput" type="text" name="name" placeholder="Item Name" value={name} onChange={event => setName(event.target.value)}/>

              <input className="formInput" type="text" name="category" placeholder="Category" value={category} onChange={event => setCategory(event.target.value)}/>

              <ImageUpload onChange={url => setPhoto(url)}/>

              <textarea className="textarea" placeholder="Item Description" name="description" value={description} onChange={event => setDescription(event.target.value)} />

              <input className="formInput" type="text" placeholder="Location" name="location" value={location} onChange={event => setLocation(event.target.value)}/>

              <button className="button primaryBtn">Add Item</button>
            </form>
          </div>
        </div>
        <Footer />
      </div>
    </div>
  )
}
export default ListAnItem
