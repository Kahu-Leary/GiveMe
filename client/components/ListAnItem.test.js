import React from 'react'

import renderer from 'react-test-renderer'

import ListAnItem from './ListAnItem'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ListAnItem />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
