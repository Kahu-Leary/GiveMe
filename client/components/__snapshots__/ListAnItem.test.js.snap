// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="itemForm"
>
  <div
    className="imgContainer"
  >
    <img
      className="cityImg"
      src="./img/city.png"
    />
  </div>
  <div
    className="formRight"
  >
    <h1>
      List a new item
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        className="formInput"
        name="name"
        onChange={[Function]}
        placeholder="Item Name"
        type="text"
        value=""
      />
      <input
        className="formInput"
        name="category"
        onChange={[Function]}
        placeholder="Category"
        type="text"
        value=""
      />
      <div
        className="ImageUpload"
      >
        <input
          name="file"
          onChange={[Function]}
          placeholder="Upload an Image"
          type="file"
        />
        <img
          src=""
          style={
            Object {
              "width": "300px",
            }
          }
        />
      </div>
      <textarea
        className="textarea"
        name="description"
        onChange={[Function]}
        placeholder="Item Description"
        value=""
      />
      <input
        className="formInput"
        name="location"
        onChange={[Function]}
        placeholder="Location"
        type="text"
        value=""
      />
      <button
        className="button primaryBtn"
      >
        Add Item
      </button>
    </form>
  </div>
</div>
`;
